Engine/platform/osx/alplmac.mm

```diff
@@ -26,3 +26,15 @@ void AGSMacGetBundleDir(char gamepath[PATH_MAX])
   strcpy(gamepath, [parentDir UTF8String]);
   }
 }
+
+int AGSMacGetFreeSpaceInMB(const char path[PATH_MAX])
+{
+  @autoreleasepool {
+  NSString* dirpath = @(path);
+  NSError *error;
+  NSDictionary* fileAttributes = [[NSFileManager defaultManager] attributesOfFileSystemForPath:dirpath
+                                                                                       error:&error];
+  unsigned long long freeSpace = [[fileAttributes objectForKey:NSFileSystemFreeSize] longLongValue];
```

Comment:
Warning in the docs... I wonder if the project would need a PrivacyInfo file:
https://developer.apple.com/documentation/foundation/nsfilesystemfreesize

Reply:
```suggestion
  unsigned long long freeSpace = [[fileAttributes objectForKey:NSFileSystemFreeSize] unsignedLongLongValue];
```

Reply:
I created a PrivacyInfo file and added in the Xcode project, I only added the NSPrivacyTracking with it set to false. From what I am getting we would only need to list everything we use if we set that to true.